Core/src/Propagator/detail/CovarianceEngine.cpp

```diff
@@ -366,8 +371,10 @@ void transportCovarianceToCurvilinear(BoundSymMatrix& boundCovariance,
 
   // Apply the actual covariance transport to get covariance of the current
   // curvilinear parameters
-  boundCovariance = fullTransportJacobian * boundCovariance *
-                    fullTransportJacobian.transpose();
+  using namespace MatrixHelpers;
+  boundCovariance =
+      multiply(fullTransportJacobian,
+               multiply(boundCovariance, transpose(fullTransportJacobian)));
```

Comment:
Transposes are both memory-bound and accessing data in the wrong order, so they are expensive unless successfully elided by the compiler. Which it may not do here due to the sheer amount of code that it is already diving through.

To avoid the transpose here, you can try either of two strategies:

- Implement the minimal composite operation that avoids the transpose, `multiplyByTranspose`. This is the route that Lucas took back in the day.
- Bite the bullet and implement the entire `multiply(T, multiply(U, transpose(T)))` round trip as a single composite operation, as previously suggested by @asalzburger .